tools/clang/unittests/HLSLExec/LongVectors.cpp

```diff
@@ -1542,6 +1542,47 @@ template <typename T> T waveMultiPrefixProduct(T A, UINT) {
   return A * A;
 }
 
+template <typename T> struct Op<OpType::WaveMatch, T, 1> : StrictValidation {};
+
+template <typename T> struct ExpectedBuilder<OpType::WaveMatch, T> {
+  static std::vector<UINT> buildExpected(Op<OpType::WaveMatch, T, 1> &,
+                                         const InputSets<T> &Inputs,
+                                         UINT WaveSize) {
+    // For this test, the shader arranges it so that lane 0 is different from
+    // all the other lanes. Besides that all other lines write their result of
+    // WaveMatch as well.
+
+    DXASSERT_NOMSG(Inputs.size() == 1);
+    std::vector<UINT> Expected;
+    Expected.assign(WaveSize * 4, 0);
+
+    UINT LowWaves = std::min(64U, WaveSize);
+    UINT HighWaves = WaveSize - LowWaves;
+
+    uint64_t LowWaveMask = (LowWaves < 64) ? (1ULL << LowWaves) : 0;
+    uint64_t HighWaveMask = (HighWaves < 64) ? (1ULL << HighWaves) : 0;
```

Comment:
Neither of these are actually storing the mask - they don't become a mask until you `- 1`.